esphome/components/lvgl/btnmatrix.py

```diff
@@ -0,0 +1,275 @@
+from esphome import automation
+import esphome.codegen as cg
+from esphome.components.key_provider import KeyProvider
+import esphome.config_validation as cv
+from esphome.const import CONF_ID, CONF_WIDTH
+from esphome.cpp_generator import MockObj
+
+from .automation import action_to_code
+from .btn import lv_btn_t
+from .defines import (
+    BTNMATRIX_CTRLS,
+    CONF_BUTTONS,
+    CONF_CONTROL,
+    CONF_ITEMS,
+    CONF_KEY_CODE,
+    CONF_MAIN,
+    CONF_ONE_CHECKED,
+    CONF_ROWS,
+    CONF_SELECTED,
+    CONF_TEXT,
+)
+from .helpers import lvgl_components_required
+from .lv_validation import key_code, lv_bool
+from .lvcode import lv, lv_add, lv_expr
+from .schemas import automation_schema
+from .types import (
+    LV_BTNMATRIX_CTRL,
+    LV_STATE,
+    LvBoolean,
+    LvCompound,
+    LvType,
+    ObjUpdateAction,
+    char_ptr,
+    lv_pseudo_button_t,
+)
+from .widget import Widget, WidgetType, get_widgets, widget_map
+
+CONF_BUTTONMATRIX = "buttonmatrix"
+CONF_BUTTONMATRIX_TEXT_LIST_ID = "buttonmatrix_text_list_id"
+
+LvBtnmBtn = LvBoolean(
+    str(cg.uint16),
+    parents=(lv_pseudo_button_t,),
+)
+BTNM_BTN_SCHEMA = cv.Schema(
+    {
+        cv.Optional(CONF_TEXT): cv.string,
+        cv.Optional(CONF_KEY_CODE): key_code,
+        cv.GenerateID(): cv.declare_id(LvBtnmBtn),
+        cv.Optional(CONF_WIDTH, default=1): cv.positive_int,
+        cv.Optional(CONF_CONTROL): cv.ensure_list(
+            cv.Schema(
+                {cv.Optional(k.lower()): cv.boolean for k in BTNMATRIX_CTRLS.choices}
+            )
+        ),
+    }
+).extend(automation_schema(lv_btn_t))
+
+BTNMATRIX_SCHEMA = cv.Schema(
+    {
+        cv.Optional(CONF_ONE_CHECKED, default=False): lv_bool,
+        cv.GenerateID(CONF_BUTTONMATRIX_TEXT_LIST_ID): cv.declare_id(char_ptr),
```

Comment:
Since this is inside a button matrix schema, do we really need the "buttonmatrix" prefix here?
```suggestion
        cv.GenerateID(CONF_TEXT_LIST_ID): cv.declare_id(char_ptr),
```

Reply:
It is a little redundant - I'll go with `CONF_BUTTON_TEXT_LIST_ID` to keep it clear what it applies to.